index.js

```diff
@@ -46,13 +60,31 @@ class Overlay extends React.Component {
   _stopPropagation = (e) => e.stopPropagation()
 
   render () {
-    const {animationType, closeOnTouchOutside, children, containerStyle, childrenWrapperStyle} = this.props;
+    const {
+      animationType,
+      onShow,
+      closeOnTouchOutside,
+      children,
+      containerStyle,
+      childrenWrapperStyle,
+      transparent,
+      hardwareAccelerated,
+      presentationStyle,
+      supportedOrientations,
+      onOrientationChange,
+    } = this.props;
     return (
       <Modal
-          animationType={animationType}
-          transparent
-          visible={this.state.visible}
-          onRequestClose={this._hideModal}>
+        animationType={animationType}
+        transparent={transparent}
+        visible={this.state.visible}
+        onRequestClose={this._hideModal}
+        onShow={onShow}
+        hardwareAccelerated={hardwareAccelerated}
```

Comment:
I think a better way would be to forward all the extra props to the modal component.

Something like 

```
const {animationType, closeOnTouchOutside, children, containerStyle, childrenWrapperStyle, ...extraProps} = this.props;
```

Then pass the extraProps to the Modal component like this:

```
<Modal {...extraProps}>
```